tests/remotes/base.py

```diff
@@ -16,7 +16,7 @@ def is_dir(self):
     def exists(self):
         raise NotImplementedError
 
-    def mkdir(self, mode=0o777, parents=False, exist_ok=False):
+    def mkdir(self, mode=0o777, parents=False, exist_ok=False, empty=False):
```

Comment:
It's pretty odd that `empty` is leaking into this fixture. How about we create separate test cases for such filesystems? They can use raw fs interface, no need to use this fixture.

Reply:
> It's pretty odd that empty is leaking into this fixture.

I didn't want to override whole `_gen()` function just to support it

> How about we create separate test cases for such filesystems?

Or perhaps we can just to it explicitly on the testcase with determining what the contents should be. If you think that it is a bit weird to do this on spot, we can always hoist the same code into `_gen()`.